src/metrics/review-coverage.ts

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import gql from 'graphql-tag';
+
+import {getOrgRepos} from '../common-queries';
+import {GitHub} from '../gql';
+import {PullRequestCommitsQuery, PullRequestCommitsQueryVariables} from '../gql-types';
+import {RepoCommitsQuery, RepoCommitsQueryVariables} from '../gql-types';
+import {RepoPRsCommitsQuery, RepoPRsCommitsQueryVariables} from '../gql-types';
+
+import {getReviewsForPullRequest} from './review-latency';
+
+export class ReviewCoverageResult {
+  commits: Map<string, ReviewedCommit>;
+
+  constructor(commits: Map<string, ReviewedCommit>) {
+    this.commits = commits;
+  }
+
+  summary() {
+    const count = this.numReviewed();
+    return `There are ${this.commits.size} commits of which ${
+        count} are reviewed. \nReview coverage is ${
+        Math.round(count / this.commits.size * 100)}%.`;
+  }
+
+  numReviewed(): number {
+    let numReviewed = 0;
+    this.commits.forEach((x) => x.reviewed ? numReviewed++ : '');
+    return numReviewed;
+  }
+}
+
+type ReviewCoverageOpts = {
+  org: string,
+  repo?: string,
+  since?: string,  // ISO date string
+};
+
+type Commit = {
+  oid: string,
+  committedDate: string,
+};
+
+type ReviewedCommit = Commit&{
+  reviewed: boolean,
+};
+
+/**
+ * Computes how much of an org/repository is reviewed.
+ */
+export async function getReviewCoverage(
+    github: GitHub, opts: ReviewCoverageOpts): Promise<ReviewCoverageResult> {
+  let repos;
+  if (opts.repo) {
+    repos = [{owner: opts.org, name: opts.repo}];
+  } else {
+    repos = await getOrgRepos(github, opts.org);
+  }
+
+  const commits: Map<string, ReviewedCommit> = new Map();
+  for (const {owner, name} of repos) {
+    // Get all commits on the master branch.
+    for (const commit of await getMasterCommits(github, owner, name)) {
+      commits.set(commit.oid, Object.assign({reviewed: false}, commit));
+    }
+
+    // Mark all reviewed PR commits as reviewed.
+    for (const commit of await getPRCommitsForRepo(github, owner, name)) {
```

Comment:
Maybe swap the order of these for loops, so you first make a `Set<string>` with all the reviewed oids? Then you could have:

`Object.assign({reviewed: reviewedCommits.has(oid)}, ...)`,  so you wouldn't need the `!`, and you don't need to set the reviewed state twice.

Reply:
Done.